code/datums/components/manual_heart.dm

```diff
@@ -0,0 +1,170 @@
+
+	//How much to heal per pump, negative numbers would HURT the player
+	var/heal_brute = 0
+	var/heal_burn = 0
+	var/heal_oxy = 0
+
+/datum/component/manual_heart/Initialize(pump_delay = 3 SECONDS, blood_loss = BLOOD_VOLUME_NORMAL * 0.2, heal_brute = 0, heal_burn = 0, heal_oxy = 0)
+	//Non-Carbon mobs can't have hearts, and should never receive this component.
+	if (!iscarbon(parent))
+		stack_trace("Manual Heart component added to [parent] ([parent?.type]) which is not a /mob/living/carbon subtype.")
+		return COMPONENT_INCOMPATIBLE
+
+	src.pump_delay = pump_delay
+	src.blood_loss = blood_loss
+	src.heal_brute = heal_brute
+	src.heal_burn = heal_burn
+	src.heal_oxy = heal_oxy
+
+	pump_action = new(src)
+	pump_action.cooldown_time = pump_delay - (1 SECONDS) //you can pump up to a second early
+	pump_action.Grant(parent)
+
+	var/mob/living/carbon/carbon_parent = parent
+	var/obj/item/organ/internal/heart/parent_heart = carbon_parent.get_organ_slot(ORGAN_SLOT_HEART)
+	if(parent_heart && !HAS_TRAIT(carbon_parent, TRAIT_NOBLOOD) && carbon_parent.stat != DEAD)
+		START_PROCESSING(SSdcs, src)
+		last_pump = world.time
+
+	to_chat(parent, span_userdanger("Your heart no longer beats automatically! You have to pump it manually - otherwise you'll die!"))
+
+/datum/component/manual_heart/Destroy()
+	to_chat(parent, span_userdanger("You feel your heart start beating normally again!"))
+	var/mob/living/carbon/carbon_parent = parent
+	if(istype(carbon_parent))
+		carbon_parent.remove_client_colour(/datum/client_colour/manual_heart_blood)
+	QDEL_NULL(pump_action)
+	return ..()
+
+/datum/component/manual_heart/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_CARBON_LOSE_ORGAN, PROC_REF(check_removed_organ))
+	RegisterSignal(parent, COMSIG_CARBON_GAIN_ORGAN, PROC_REF(check_added_organ))
+	RegisterSignals(parent, list(COMSIG_LIVING_DEATH, SIGNAL_ADDTRAIT(TRAIT_NOBLOOD)), PROC_REF(pause))
+	RegisterSignals(parent, list(COMSIG_LIVING_REVIVE, SIGNAL_REMOVETRAIT(TRAIT_NOBLOOD)), PROC_REF(restart))
+
+/datum/component/manual_heart/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_CARBON_GAIN_ORGAN, COMSIG_CARBON_LOSE_ORGAN, COMSIG_LIVING_REVIVE, COMSIG_LIVING_DEATH, SIGNAL_ADDTRAIT(TRAIT_NOBLOOD), SIGNAL_REMOVETRAIT(TRAIT_NOBLOOD)))
+
+/datum/component/manual_heart/proc/restart()
+	SIGNAL_HANDLER
+
+	if(check_valid())
+		last_pump = world.time
+		pump_action.build_all_button_icons(UPDATE_BUTTON_STATUS) //make sure the action button always shows as available when it is
+		START_PROCESSING(SSdcs, src)
+
+/datum/component/manual_heart/proc/pause()
+	SIGNAL_HANDLER
+	pump_action.build_all_button_icons(UPDATE_BUTTON_STATUS)
+	STOP_PROCESSING(SSdcs, src)
+
+/// Worker proc that checks logic for if a pump can happen, and applies effects/notifications from doing so
+/datum/component/manual_heart/proc/on_pump(mob/owner)
+	last_pump = world.time
+	playsound(owner,'sound/effects/singlebeat.ogg', 40, TRUE)
+	owner.balloon_alert(owner, "your heart beats")
```

Comment:
Are we sure we want to show this balloon alert every 2-3 seconds?

Reply:
I thought it would be good to have some feedback that it actually worked, other than the very soft heartbeat sound - but I can remove the balloon alert if you think that's better.

Reply:
I think the sound alone is enough, the frequency of this alert I think would be a bit too much